src/aeserialization.erl

```diff
@@ -117,6 +125,12 @@ encode_field(Type, Val) -> error({illegal, Type, Val}).
 
 decode_field([Type], List) when is_list(List) ->
     [decode_field(Type, X) || X <- List];
+decode_field(#{items := Items}, List) when length(List) =:= length(Items) ->
+    Zipped = lists:zip(Items, List),
+    lists:foldl(
+      fun({{K, Type}, V}, Map) ->
+              Map#{ K => decode_field(Type, V) }
+      end, #{}, Zipped);
```

Comment:
Check for duplicates in `Items`?
```suggestion
decode_field(#{items := Items}, List) when length(List) =:= length(Items) ->
    Zipped = lists:zip(Items, List),
    lists:foldl(
      fun({{K, Type}, V}, Map) ->
              maps:is_key(K, Map) andalso error(badarg, duplicate_field),
              Map#{ K => decode_field(Type, V) }
      end, #{}, Zipped);
```

Reply:
> Check for duplicates in `Items`?

I guess we could, but there will be runtime overhead to catch what would be a compile-time error.